mongo.go

```diff
@@ -40,12 +41,19 @@ func newMongoItem(i *resource.Item) *mongoItem {
 func newItem(i *mongoItem) *resource.Item {
 	// Add the id back (we use the same map hoping the mongoItem won't be stored back)
 	i.Payload["id"] = i.ID
-	return &resource.Item{
+	item := &resource.Item{
 		ID:      i.ID,
 		ETag:    i.ETag,
 		Updated: i.Updated,
 		Payload: i.Payload,
 	}
+
+	if item.ETag == "" {
+		if v, ok := i.ID.(bson.ObjectId); ok {
+			item.ETag = "W/" + v.Hex()
```

Comment:
Change to `P/` for "preliminary" over `W/` for "weak"?

Reply:
Okay, if I put `P/` then this goes outside HTTP `ETag` standart. It is fine with me, but don't know about Olivier.

Reply:
Maybe `P/` is not the best neither - maybe `prelim-` or something else - we just need to make it _recognisable_. 

Seen from the perspective of the browser, the preliminary `E-tag` _should be_ no-less strong then the MD5 sum. I.e. if the data, once imported and "transferred" to being managed by rest-layer, is not modified outside of rest-layer without also generating/updating the E-tag, there is no (more) ways the data _could change_ without the E-tag being regenerated, compared to an item that has the E-tag set in the DB. Therefore _if_ the payload of an item with an E-tag from the DB is byte-compatible, one with a preliminary E-tag should be equally byte-compatible for a unique URL / Query string combination.

Thus if we want to add a `W/` for "weak", we _should_ probably do it for _all_ E-tags. This is probably something we wold want to add in the _representation_ layer, rather than the storage later, so not part of this PR. Although the item payloads would only end up _not_ being completely byte-compatible in a few edge-cases based on the `resource`, `rest` and `schema` packages , as I understand it from your link, the actual use-case for _strong_ E-tags is for e.g. continuing an interrupted download, which does not strike me as normal REST API usage patterns.

Reply:
On another subject, this only generates an E-tag if the ID is an Object-ID. Should we not also support other ID formats? E.g. fallback to a `fmt.Sprint(i.ID)`?

Reply:
If we are going that route (all weak Etag), we should use some form of special Etag formatting to recognize the missing Etag and not break the Etag spec by inventing new prefix (i.e.: The customizable part is inside the quotes).

Reply:
I am for making all ETags weak, and adding that prefix in the `rest-layer` itself, and not in storage.
I guess we should just pick a prefix for the missing `_etag` case. Like `v0-1234`, or `p-1234` where `1234` is the doc ID. If we even choose to change this later, it will be transparent for the main logic.

@smyrman Good catch, I will add fallback when ID is not `ObjectID`.

Reply:
So `"<prefix>-<id>"` or something.

Reply:
Perfect!

Reply:
Lets see what @rs will say about `ETag` prefix, and I will update the PR.

Reply:
Works for me